### DISCUSSION TITLE
Help creating screen with variable fields

### DISCUSSION THREAD

**Discussion by cyber-kivi** [Budibase/budibase#6393] [Help]
Hello there!
Please, help creating screen with variable fields (Headline and enter Text field depend from selected row):

0. Main table with data
1. Select row (link)
2. Open screen for editing attributes for the selected row 
It contains 
 - Headline (from directory for this data row)
 - Text field
Condition: a row can have >=1 attribute. Depending on the selected row, the list of attributes changes. Three attributes equals three Headline and three text fields
4. Enter attribute value into the text field

![image](https://user-images.githubusercontent.com/106427850/174598319-92ec305b-711a-4bc0-8251-80ea2313d95b.png)
 I have directory of rows - DirNosEd
 and for atributes - DirAtr

It looks like editing row, but I can't make some variable fields

**Comment by melohagan** [comment]
Hi @cyber-kivi 

The generated "Edit" screens by default will include all available attributes of the main table.

You could try using Conditional UI (https://docs.budibase.com/docs/conditions)
If the value is blank, then hide the field.

**Reply by cyber-kivi** [reply]
Thanks for your quick reply!

I tried to use Conditional, and this "If the value is blank, then hide the field." - maybe I did something wrong or it doesn't work.

This is what I need:
1. Select "FP"
![1p](https://user-images.githubusercontent.com/106427850/174807301-caf82b54-72a2-46b6-9b6f-35045414d58e.png)

2. Show only "FP" attributes
![2p](https://user-images.githubusercontent.com/106427850/174807729-812f4436-d4a9-4c2f-805c-5f4e8b854cfd.png)

Filtering by name in "main" table (NosEdPat) to get attributes from another table (DirAtr)
![3](https://user-images.githubusercontent.com/106427850/174809064-d8d9d9ff-0061-4a36-8efc-c96f390e7bda.png)
![4](https://user-images.githubusercontent.com/106427850/174809112-095346e8-90c8-4751-b837-cf95aa4067dc.png)

How can I make it?

**Reply by melohagan** [reply]
Considering you are clicking a link to open in a new screen, I don't see a reason this wouldn't work with a filtered Data Provider and repeater (similar to the autogenerated screens):
https://docs.budibase.com/docs/url-parameters
